pkgs/dart_tooling_mcp_server/test/tools/dart_cli_test.dart

```diff
@@ -2,68 +2,39 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'dart:io';
-
 import 'package:dart_mcp/server.dart';
 import 'package:dart_tooling_mcp_server/src/mixins/dart_cli.dart';
-import 'package:path/path.dart' as p;
 import 'package:test/test.dart';
 
 import '../test_harness.dart';
 
 void main() {
   late TestHarness testHarness;
+  late TestProcessManager testProcessManager;
+
+  // This root is arbitrary for these tests since we are not actually running
+  // the CLI commands, but rather sending them through the
+  // [TestProcessManager] wrapper.
+  final testRoot = rootForPath(counterAppPath);
 
   // TODO: Use setUpAll, currently this fails due to an apparent TestProcess
   // issue.
   setUp(() async {
-    testHarness = await TestHarness.start();
+    testHarness = await TestHarness.start(inProcess: true);
+    testProcessManager =
+        testHarness.serverConnectionPair.server!.processManager
+            as TestProcessManager;
+    addTearDown(() => testProcessManager.reset());
```

Comment:
Should we just call `addTearDown(reset)` in the constructor of `TestProcessManager` to do this?